packages/app/src/app/components/CreateSandbox/Import/importLimits.tsx

```diff
@@ -0,0 +1,152 @@
+import track from '@codesandbox/common/lib/utils/analytics';
+import {
+  Element,
+  Link as StyledLink,
+  MessageStripe,
+} from '@codesandbox/components';
+import { SUBSCRIPTION_DOCS_URLS } from 'app/constants';
+import { useGetCheckoutURL } from 'app/hooks/useCreateCheckout';
+import { useSubscription } from 'app/hooks/useSubscription';
+import { useWorkspaceAuthorization } from 'app/hooks/useWorkspaceAuthorization';
+import { useActions, useAppState } from 'app/overmind';
+import React from 'react';
+import { Link as RouterLink, useLocation } from 'react-router-dom';
+
+const getEventName = (
+  isEligibleForTrial: boolean,
+  isTeamAdmin?: boolean
+): string => {
+  if (isTeamAdmin) {
+    return isEligibleForTrial
+      ? 'Limit banner: import - Start Trial'
+      : 'Limit banner: import - Upgrade';
+  }
+
+  return 'Limit banner: import - Learn more';
+};
+
+export const MaxPublicRepos: React.FC = () => {
+  const { activeTeam } = useAppState();
+  const { isEligibleForTrial } = useSubscription();
+  const { isTeamAdmin } = useWorkspaceAuthorization();
+  const { pathname } = useLocation();
+  const { modals } = useActions();
+
+  const checkout = useGetCheckoutURL({
+    team_id: isTeamAdmin ? activeTeam : undefined,
+    success_path: pathname,
+    cancel_path: pathname,
+  });
+
+  return (
+    <MessageStripe justify="space-between">
+      You&apos;ve reached the maximum amount of free repositories. Upgrade for
+      more.
+      {isTeamAdmin ? (
```

Comment:
this logic is reused quite a while, don't want to push for over abstraction but maybe there's room  for a `SubscribeToProMessageStripe` that only gets the message we want to show and some other props maybe and determines internally if it shows the CTA or the link to docs

Reply:
Not suggesting this for the current PR, rather for a follow-up refactor